src/core/exec-invoke.c

```diff
@@ -4212,6 +4212,166 @@ static bool exec_context_need_unprivileged_private_users(
                !strv_isempty(context->no_exec_paths);
 }
 
+static bool exec_namespace_is_delegated(const ExecContext *context, const ExecParameters *params, unsigned long namespace) {
+        /* If we need unprivileged private users, we've already unshared a user namespace by the time we call
+         * setup_namespaces() for the first time so let's make sure we do all other namespace unsharing in
+         * the first call to setup_namespaces() by returning false here. */
+        if (exec_context_need_unprivileged_private_users(context, params))
+                return false;
+
+        if (context->private_users == PRIVATE_USERS_NO)
+                return false;
+
+        if (context->delegate_namespaces == NAMESPACE_FLAGS_INITIAL)
+                return false;
+
+        return FLAGS_SET(context->delegate_namespaces, namespace);
+}
+
+static int setup_namespaces(
+                const ExecContext *context,
+                ExecParameters *params,
+                ExecRuntime *runtime,
+                const char *memory_pressure_path,
+                uid_t uid,
+                uid_t gid,
+                const ExecCommand *command,
+                bool needs_sandboxing,
+                bool has_cap_sys_admin,
+                bool delegate,
+                int *reterr_exit_status) {
+
+        int r;
+
+        if (exec_needs_network_namespace(context) &&
+            exec_namespace_is_delegated(context, params, CLONE_NEWNET) == delegate &&
+            runtime && runtime->shared && runtime->shared->netns_storage_socket[0] >= 0) {
+
+                /* Try to enable network namespacing if network namespacing is available and we have
+                 * CAP_NET_ADMIN. We need CAP_NET_ADMIN to be able to configure the loopback device in the
+                 * new network namespace. And if we don't have that, then we could only create a network
+                 * namespace without the ability to set up "lo". Hence gracefully skip things then. */
```

Comment:
please emphasize in the comment that we care about CAP_NET_ADMIN in the userns we are running in *at* *this* *point*, which could be thw host's or the fresh one.